### PR TITLE
Updated to pika 1.0 (from beta)

### PR DESCRIPTION
**Pull request by muhrin**
Had to fix one test and took the opportunity to remove an unused
dependency on YAML and fixed up some prospector problems in test_amqp

### CODE REVIEW

File: test/test_amqp.py [hunk 1]
```diff
@@ -654,7 +656,7 @@ def test_purge_queue(self):
 
             with self.assertRaises(gen.TimeoutError):
                 yield queue.get(timeout=1)
-        except:
+        except topika.exceptions.QueueEmpty:
```
Comment (greschd) [inline test/test_amqp.py]: Does the timeout also raise an exception here?

File: test/test_amqp.py [hunk 2]
```diff
@@ -848,8 +855,8 @@ def test_add_close_callback(self):
 
         shared_list = []
 
-        def share(f):
-            shared_list.append(f)
+        def share(_):
```
Comment (greschd) [inline test/test_amqp.py]: I don't think ``_`` should be used here - the convention (as I know it) is for ``_`` to be a "throwaway" variable that is never used.

File: test/test_amqp.py (hunk 1 above)
Comment (greschd) [inline test/test_amqp.py]: Is this just generic cleanup, or really something that only should happen when the queue is empty?

If it's the former, ``try .. finally`` would be more appropriate.

Comment (muhrin) [reply test/test_amqp.py]: Thanks.  I did `finally` first but indeed it doesn't work.

File: test/test_amqp.py (hunk 2 above)
Comment (muhrin) [reply test/test_amqp.py]: Roger.  I'll choose a more reasonable name...

Comment (muhrin) [reply test/test_amqp.py]: Done.

**Pull request merged**